### PR TITLE
enable -Wsuggest-override

### PR DESCRIPTION
**Pull request by mikea**
Having override annotations in place greatly helps with API
refactorings. I also believe this is a commonly accepted coding
practice.

### CODE REVIEW

**Review by jasnell — APPROVED** [review]

File: c++/.bazelrc [hunk 1]
```diff
@@ -2,12 +2,14 @@ common --noenable_bzlmod
...
-build:unix --cxxopt='-Wall' --host_cxxopt='-Wall'
-build:unix --cxxopt='-Wextra' --host_cxxopt='-Wextra'
+build:unix --cxxopt='-Wall'
+build:unix --cxxopt='-Wextra'
+build:unix --cxxopt='-Wsuggest-override'
 build:unix --cxxopt='-Wno-strict-aliasing' --host_cxxopt='-Wno-strict-aliasing'
 build:unix --cxxopt='-Wno-sign-compare' --host_cxxopt='-Wno-sign-compare'
 build:unix --cxxopt='-Wno-unused-parameter' --host_cxxopt='-Wno-unused-parameter'
 build:unix --cxxopt='-Wno-deprecated-this-capture' --host_cxxopt='-Wno-deprecated-this-capture'
+build:unix --cxxopt='-Wno-deprecated-this-capture' --host_cxxopt='-Wno-deprecated-this-capture'
```
Comment (fhanau) [inline c++/.bazelrc:12]: Nit: duplicate line

**Comment by fhanau** [comment]
Figured out why the super-test build fails with a suggest-override warning turned into an error, while the bazel build doesn't even emit the warning: The bazel build fails to enable `-DCAPNP_HEADER_WARNINGS`, causing the warning to not be emitted with bazel. Since the warning is in a generated capnp header we might need changes to the capnp code generator to fully address this.

**Comment by mikea** [comment]
> Figured out why the super-test build fails with a suggest-override warning turned into an error, while the bazel build doesn't even emit the warning: The bazel build fails to enable `-DCAPNP_HEADER_WARNINGS`, causing the warning to not be emitted with bazel. Since the warning is in a generated capnp header we might need changes to the capnp code generator to fully address this.

awesome, this indeed helps. I have added `CAPNP_HEADER_WARNINGS` to `.bazelrc`

**Comment by mikea** [comment]
I tried to bring bazel behavior closer to super-test here: https://github.com/capnproto/capnproto/pull/2045 @fhanau

**Review by fhanau — APPROVED** [review]

**Comment by kentonv** [comment]
Oh cool, if I'd known about this flag I would have enabled it a long time ago!

**Pull request merged**